### PR TITLE
AerodromeConnector Enhancements and Optimizations

### PR DESCRIPTION
**Pull request by njokuScript**
### Overview
This PR introduces several significant enhancements and optimizations to the AerodromeConnector, improving its functionality, efficiency, and test coverage.

### Key Changes
1. Liquidity Management Improvements
2. Updated `LIQ_SLIPPAGE` constants for more accurate slippage control
3. Implemented staking of LP tokens
4. Fixed deposit gauge functionality and cleaned up connectors

#### Testing and Security
- Added tests for locking of LP tokens, enhancing security and reliability

### Impact
These changes significantly enhance the AerodromeConnector's capabilities, particularly in areas of liquidity provision and staking of LP tokens.

### CODE REVIEW

File: src/connectors/base/aerodrome/interface.sol [hunk 1]
```diff
@@ -8,6 +8,15 @@ interface IWETH {
...
+    function withdraw(uint256 _amount) external;
+    function getReward(address _account) external;
+    function rewardToken() external view returns (address);
+    function balanceOf(address) external view returns (uint256);
+    function stakingToken() external view returns (address);
+}
+
 /// @title Router token swapping functionality
 /// @notice Functions for swapping tokens via CL
 interface ISwapRouterV3 {
```
Comment (MeisterJustice) [inline src/connectors/base/aerodrome/interface.sol:22]: You forgot to include an interface description for IGauge

File: src/connectors/base/aerodrome/main.sol [hunk 2]
```diff
@@ -216,4 +222,64 @@ contract AerodromeConnector is BaseConnector, Constants, AerodromeEvents {
...
+
+    //     // Transfer claimed fees to the caller
+    //     if (claimed0 > 0) {
+    //         IERC20(token0).transfer(msg.sender, claimed0);
+    //     }
+    //     if (claimed1 > 0) {
+    //         IERC20(token1).transfer(msg.sender, claimed1);
+    //     }
+
+    //     emit FeesWithdrawn(poolAddress, token0, claimed0, token1, claimed1);
```
Comment (MeisterJustice) [inline src/connectors/base/aerodrome/main.sol:282]: If this will be handled in the SDK, shouldn't we remove it? rather tan having commented piece of code

**Review by MeisterJustice — CHANGES_REQUESTED** [review]

File: src/connectors/base/aerodrome/main.sol (hunk 2 above)
Comment (njokuScript) [reply src/connectors/base/aerodrome/main.sol:282]: true. will update!

File: src/connectors/base/aerodrome/interface.sol (hunk 1 above)
Comment (njokuScript) [reply src/connectors/base/aerodrome/interface.sol:22]: great catch!

File: src/connectors/base/common/constant.sol [hunk 3]
```diff
@@ -20,8 +20,9 @@ abstract contract Constants {
     address internal constant AERODROME_FACTORY = 0x420DD381b31aEf6683db6B902084cB0FFECe40Da;
 
     /// @dev Liquidity slippage tolerance: 0.5%
-    // 10% for testing
-    uint256 internal constant LIQ_SLIPPAGE = 1000;
+    // 3% for testing
+    // 10000 = 100%, 5000 = 50%, 100 = 1%, 1 = 0.01%
+    uint256 internal constant LIQ_SLIPPAGE = 300;
```
Comment (renanmav) [inline src/connectors/base/common/constant.sol:25]: Every change here requires a new contract deployment, right?

Do we want to control this in another way in the future? Maybe configs through metadata?

Also, line 22 says slippage tolerance is 0.5%, isn't that outdated info?

Comment (njokuScript) [reply src/connectors/base/common/constant.sol:25]: nice catch! will update line 22 and also make the constants that could change like `LIQ_SLIPPAGE`

Comment (njokuScript) [reply src/connectors/base/common/constant.sol:25]: you can raise a pr for this later? not a priority @renanmav

**Review by MeisterJustice — APPROVED** [review]

File: src/connectors/base/aerodrome/interface.sol [hunk 4]
```diff
@@ -8,6 +8,17 @@ interface IWETH {
     function withdraw(uint256 amount) external;
 }
 
+/// @title LP staking functionality
+/// @notice Gauge interface for interacting with guage contract for distribution of emissions by address
```
Comment (renanmav) [inline src/connectors/base/aerodrome/interface.sol:12]: NIT: small typo

```suggestion
/// @notice Gauge interface for interacting with gauge contract for distribution of emissions by address
```

**Review by renanmav — APPROVED** [review]
LGTM

**Pull request merged**